torch/utils/data/distributed.py

```diff
@@ -1,15 +1,12 @@
 import math
-from typing import TypeVar, Optional, Iterator
+from typing import Optional, Iterator
 
 import torch
 from . import Sampler, Dataset
 import torch.distributed as dist
 
 
-T_co = TypeVar('T_co', covariant=True)
-
-
-class DistributedSampler(Sampler[T_co]):
+class DistributedSampler(Sampler[int]):
```

Comment:
Any reason why the typing here is changed?

Reply:
When switching from returning the list iterator to creating a generator, it makes the type more specific.